### PR TITLE
feat(monitors): Add env filter to stats

### LINKED ISSUE #45020: Crons: Update `monitor_stats` to filter on environment
Straight forward

### PR DESCRIPTION
**Pull request by rjo100**
Will require a frontend change to see effects as stats query is generated client side and not from the page URL.

Closes https://github.com/getsentry/sentry/issues/45020

### CODE REVIEW

**Review by evanpurkhiser — APPROVED** [review]

File: src/sentry/monitors/endpoints/organization_monitor_stats.py [hunk 1]
```diff
@@ -33,8 +34,16 @@ def get(self, request: Request, organization, project, monitor) -> Response:
...
             date_added__gt=args["start"],
             date_added__lte=args["end"],
-        ).values_list("date_added", "status", "duration")
-        for datetime, status, duration in history.iterator():
+        )
+
+        environments = get_environments(request, organization)
+
+        if environments:
+            history = history.filter(monitor_environment__environment__in=environments)
```
Comment (evanpurkhiser) [inline src/sentry/monitors/endpoints/organization_monitor_stats.py:42]: When filter on multiple environments what will this all look like?

Comment (rjo100) [reply src/sentry/monitors/endpoints/organization_monitor_stats.py:42]: see the test case. right now it presents them aggregated. floated this earlier. think we should keep it simple for now to ship

Comment (evanpurkhiser) [reply src/sentry/monitors/endpoints/organization_monitor_stats.py:42]: 👍

**Review by evanpurkhiser — APPROVED** [review]

**Review by wedamija — APPROVED** [review]
Makes sense to me, we can work out what splitting these up looks like later

**Pull request merged**